taskcluster/fxci_config_taskgraph/optimizations.py

```diff
@@ -0,0 +1,44 @@
+import os
+import subprocess
+from functools import cache
+
+from taskgraph.optimize.base import OptimizationStrategy, register_strategy
+from taskgraph.optimize.strategies import SkipUnlessChanged
+
+from fxci_config_taskgraph.util.integration import FIREFOXCI_ROOT_URL
+
+
+@register_strategy("integration-test")
+class IntegrationTestStrategy(OptimizationStrategy):
+    @cache
+    def _get_modified_worker_pools(self) -> list[str]:
+        cmd = [
+            "tc-admin",
+            "diff",
+            "--environment",
+            "firefoxci",
+            "--resources",
+            "worker_pools",
+            "--ids-only",
+        ]
+        env = os.environ.copy()
+        env["TASKCLUSTER_ROOT_URL"] = FIREFOXCI_ROOT_URL
+        proc = subprocess.run(cmd, capture_output=True, text=True, env=env)
+        lines = [line for line in proc.stdout.splitlines() if line.startswith("!")]
+
+        for i, line in enumerate(lines):
+            line = line.split()[1]
+            assert line.startswith("WorkerPool=")
+            line = line.split("=", 1)[1]
+            lines[i] = line
```

Comment:
nit: This feels like a foot gun. A different variable for storing modified worker pools would be cleaner and easier to debug.